### DISCUSSION TITLE
Making specific modules

### DISCUSSION THREAD

**Discussion by jrhoads20** [cockpit-project/cockpit#18428] [Q&A] [answered]
Is there a way when I run make to exclude certain node_modules from being made? Such as if I don't want Applications node_module to be done with Makefile, is it possible to comment out code in the Makefile, compile_commands.json, or webpack.config.js (one or all) to make this change?

**Comment by jelly** [comment, marked as answer]
If I understand you correctly and you only want to work on a certain page for example `kdump`, then after the initial `make` you can run `./tools/webmake-make kdump`. However make should really only re-compile what was changed.

**Reply by jrhoads20** [reply]
> If I understand you correctly and you only want to work on a certain page for example `kdump`, then after the initial `make` you can run `./tools/webmake-make kdump`. However make should really only re-compile what was changed.

Not exactly, to clarify my initial ask: I'm trying to run make and it takes so long to complete each time following a make clean, each time I run make there are pages compiled such as playground, storaged, and sosreport that I am not using. So each time I do a full make, I need to go in an remove these from /dist which is symlinked to /usr/share/cockpit. I guess I figured if there was a way to have these not be compiled, it would both speed up make and prevent the pages from being created at the same time.

**Reply by martinpitt** [reply]
That's exactly what @jelly was saying. If you want to work on a particular page, use e.g. `tools/webpack-watch kdump` (or whichever other page you want to work on). Then recompiling after changing a source file is a matter of a few seconds.

No need to remove files from dist/, though. You can symlink either the whole dist/ folder or individual pages in it to /usr/local/share/cockpit/ or ~/.local/share/cockpit. **Never** change things in /usr/share please, that's package manager territory.

Please see https://github.com/cockpit-project/cockpit/blob/main/HACKING.md#working-on-your-local-machine-cockpits-session-pages

**Reply by jrhoads20** [reply]
This is a barebones system, nothing is installed except Cockpit (compiled from source), gcc and the tools to develop it a bit. The only files in the /usr/share are those symlinked from the compiled Cockpit dist folder,ie:/app/cockpit/dist ---> /usr/share/cockpit, and as I don't want some of the modules that are being built by make and added to /dist, I have been up to this point just moving them to hidden, ie "mv Playground .Playground" and it's been effective thus far. I didn't think of symlinking the individual pages, as the section you linked is what I followed and it outlined symlinking as I have it above.

My takeaway here is that I needed to be a little more creative in my thinking, use the modular tooling, and that I don't need to make clean. (Just read how often I should be doing it).

Thanks Jelly, thanks Martin!
